File: Competition/2019/Blue/Lets-Make-A-Call/solution.py


def main():
    
    letter_n_num = {
        "A" : 2,
        "B" : 2,
        "C" : 2,
        "D" : 3,
        "E" : 3,
        "F" : 3,
        "G" : 4,
        "H" : 4,
        "I" : 4,
        "J" : 5,
        "K" : 5,
        "L" : 5,
        "M" : 6,
        "N" : 6,
        "O" : 6,
        "P" : 7,
        "Q" : 7,
        "R" : 7,
        "S" : 7,
        "T" : 8,
        "U" : 8,
        "V" : 8,
        "W" : 9,
        "X" : 9,
        "Y" : 9,
        "Z" : 9
    }

    final_phone_num = ""

    print("Enter the phone number:")
    user_phone_num = input()

    for char in user_phone_num:

        if char.isalpha():

            final_phone_num += str(letter_n_num[char.upper()])
        
        else:
            final_phone_num += char
    
    print(final_phone_num)

main()

File: Competition/2019/Blue/Lets-Make-A-Call/Solution.java
import java.util.HashMap;
import java.util.Scanner;
import java.lang.StringBuilder;

public class Solution
{
    public static void main(String[] args)
    {
        HashMap<Character, Integer> charNNums = new HashMap<Character, Integer> ();
        charNNums.put('A', 2);
        charNNums.put('B', 2);
        charNNums.put('C', 2);
        charNNums.put('D', 3);
        charNNums.put('E', 3);
        charNNums.put('F', 3);
        charNNums.put('G', 4);
        charNNums.put('H', 4);
        charNNums.put('I', 4);
        charNNums.put('J', 5);
        charNNums.put('K', 5);
        charNNums.put('L', 5);
        charNNums.put('M', 6);
        charNNums.put('N', 6);
        charNNums.put('O', 6);
        charNNums.put('P', 7);
        charNNums.put('Q', 7);
        charNNums.put('R', 7);
        charNNums.put('S', 7);
        charNNums.put('T', 8);
        charNNums.put('U', 8);
        charNNums.put('W', 9);
        charNNums.put('X', 9);
        charNNums.put('Y', 9);
        charNNums.put('Z', 9);

        StringBuilder myStrBuilder = new StringBuilder();

        String finalPhoneNum = "";

        Scanner myScanner = new Scanner(System.in);

        System.out.println("Enter phone number:");
        String userPhoneNum = myScanner.nextLine();

        for (char lilChar : userPhoneNum.toCharArray())
        {
            if (Character.isAlphabetic(lilChar))
            {
                int phoneDigit = charNNums.get(Character.toUpperCase(lilChar));
                myStrBuilder.append(phoneDigit);
            }

            else
            {
                myStrBuilder.append(lilChar);
            }
        }

        finalPhoneNum = myStrBuilder.toString();

        System.out.println(finalPhoneNum);
        

    }
}